scripts/globals/pasteventwatcher.lua

```diff
@@ -0,0 +1,220 @@
+-----------------------------------
+-- Past Event Watcher NPCs
+-----------------------------------
+
+-----------------------------------
+
+require("scripts/globals/quests")
+require("scripts/globals/missions")
+require("scripts/globals/zone")
+
+pasteventwatcher = {}
+
+local gobCS = -- add the goblin cs to this table
+{
+    [102] = 126, -- La Theine Plateau
+    [106] = 247, -- North Gustaberg
+    [107] = 905, -- South Gustaberg
+    [119] = 35,  -- Meriphataud Mountains
+    [122] = 6,   -- Ro'Maeve
+    [162] = 10,  -- Castle Zvahl Keep
+    [167] = 100, -- Bostaunieux Oubliette
+    [198] = 67,  -- Maze of Shakhrami
+    [200] = 61,  -- Garlaige Citadel
+}
```

Comment:
Is there a reason we're not just passing the csid through the function?

If there is a reason this is needed, can we do away with magic numbers please?

```lua
local cutscenes = 
{
    [dsp.zone.LA_THEINE_PLATEAU] = 126,
    [dsp.zone.NORTH_GUSTABERG] = 247,
    ...etc
}
```

Reply:
I helped with these functions and I put these here so the startEvent function could send the GOBLIN_SLIPPED_AWAY text and so all the npc files would be uniform, as well as having all the csid's related to this feature in the same place.

Reply:
> and I put these here so the startEvent function could send the GOBLIN_SLIPPED_AWAY text
I'm not following, what is stopping you from doing this without the table?

Reply:
You could, but then the npc file would need to be something like
```lua
if csid ~= gobcs then
    player:messageSpecial(ID.text.GOBLIN_SLIPPED_AWAY)
else
    dsp.goblinfootprint.startEvent(player, gobcs, option)
end
```
where gobcs is different for every npc file. I figured keeping the npc files as simple as possible would be worth tabling these id's here.

Reply:
I've also since rewritten that function as
```lua
function dsp.goblinfootprint.startEvent(player, csid, option)
    local zone = player:getZoneID()
    local ID = zones[zone]
    if csid == gobCS[zone] and cutscene[zone][option] ~= nil then
        player:startEvent(unpack(cutscene[zone][option]))
        return
    else
        player:messageSpecial(ID.text.GOBLIN_SLIPPED_AWAY)
    end
end
```
so that the text message is displayed for the goblin cs as well if the player decides to cancel.

Reply:
removed magic numbers.